backend/controller/async/async.go

```diff
@@ -0,0 +1,85 @@
+package async
```

Comment:
These types had to be pulled out sooner rather than later to break an import cycle.